pandas/io/parsers.py

```diff
@@ -786,17 +786,25 @@ def read(self, nrows=None):
             return ret
 
         # May alter columns / col_dict
-        index, columns, col_dict = self._create_index(ret)
+        index, columns, mappings, col_dict = self._create_index(ret)
```

Comment:
Okay, first: could you explain where the confusion is coming from?  Telling me it's a bad solution (with no suggestions on how to proceed especially) does not provide much useful feedback.